include/seastar/core/memory.hh

```diff
@@ -139,7 +139,23 @@ static constexpr size_t huge_page_size =
 void configure(std::vector<resource::memory> m, bool mbind,
         std::optional<std::string> hugetlbfs_path = {});
 
-void enable_abort_on_allocation_failure();
+/**
+ * Set the global state of the abort on allocation failure behavior.
+ *
+ * If enabled, an allocation failure (i.e., the requested memory
+ * could not be allocate even after reclaim was attempted), will
+ * generally result in abort() being called. If disabled, the
+ * failure is reported to the caller, e.g., by throwing a
+ * std::bad_alloc for C++ allocations such as new, or returning
+ * a null pointer from malloc.
+ *
+ * Note that even if the global state is set to enabled, the 
+ * disable_abort_on_alloc_failure_temporarily class may override
+ * the behavior tepmorarily on a given shard. That is, abort only
+ * occurs if abort is globablly enabled on this shard and there
+ * are no disable_abort_... objets currently alive on this shard.
+ */
+void enable_abort_on_allocation_failure(bool enabled = true);
```

Comment:
If the call allows setting the parameter both ways, it's better to name it `set_abort_on_allocation_failure()`

Reply:
@xemul - I agree and considered this, but this would be a breaking change to the API right?

I also considered just adding another method `disable_abort_on_allocation_failure`, though I find this API a bit worse, it doesn't have the naming problem.

I'll go whatever way you think is best.

Reply:
> @xemul - I agree and considered this, but this would be a breaking change to the API right?

Ah, I wasn't clear enough, sorry :) Let me try to clarify my point.

This comment and next one should have been together -- the old API call should _not_ get removed, but it should be _deprecated_. Then we wait for ~1 year, then remove it completely.

Reply:
Thanks for the clarification @xemul, it makes sense. I'll wait for the resolution of the discussion on whether we need this at all with @raphaelsc before proceeding.

Reply:
@xemul - I've updated this PR to add a new `set_` method as you suggest and deprecate the existing one-way API.